kustomization/components/redis/README.md

```diff
@@ -0,0 +1 @@
+Sets up Redis for paperless. A redis server is required for Paperless.
```

Comment:
I kinda prefer to just include the deployment like this for tests; the component itself can just stay in your repository, at least until I add the operator stuff here and we can just add a resource here.

Reply:
I think your operator stuff can replace the component implementation. Can also treat this as an optional component then. As long as I get a service in the same namespace named 'redis', then I think it's fine.

Reply:
It won't be called `redis`, so you probably aren't going to use it.  For example, I deployed with sentinels for authelia, and ended up with this from the operator:
![image](https://github.com/marinatedconcrete/config/assets/656602/473924a0-709e-4aa0-9137-eb77b4af48cb)

Reply:
Ok, let's go with a generic name for now and we can update/patch it if/when you upstream redis. Looks like we'll have multiple modes and thus names with that operator. We should figure out if only one mode makes sense for this application. I can either rename the redis component in this PR to something like 'redis-simple' or we just come back to this in the future when we have a second Redis use case.